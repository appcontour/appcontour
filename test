#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
