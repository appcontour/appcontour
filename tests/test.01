./contour applyrule n1 <examples/tre_sfere.morse
