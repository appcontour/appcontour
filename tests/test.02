#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
