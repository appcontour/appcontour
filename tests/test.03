./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
