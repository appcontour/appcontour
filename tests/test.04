./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
