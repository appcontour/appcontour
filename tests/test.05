./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
