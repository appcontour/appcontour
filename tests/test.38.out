link L8a21: 4 components
Component 1, alexander: 1;
Component 2, alexander: 1;
Component 3, alexander: 1;
Component 4, alexander: 1;
link L10a108: 2 components
Component 1, alexander: +1-t+t^2;
Component 2, alexander: +1-t+t^2;
link L10n65: 3 components
Component 1, alexander: 1;
Component 2, alexander: 1;
Component 3, alexander: -1+3t-t^2;
link L11n459: 4 components
Component 1, alexander: 1;
Component 2, alexander: 1;
Component 3, alexander: 1;
Component 4, alexander: 1;
link L11n412: 3 components
Component 1, alexander: 1;
Component 2, alexander: 1;
Component 3, alexander: +1-t+t^2;
link L11a385: 3 components
Component 1, alexander: 1;
Component 2, alexander: 1;
Component 3, alexander: +2-3t+2t^2;
knot L11n257: abelianized jacobian matrix
-w^(-1)+2w,	+w^(-3)-w^(-1),	0,	0,	0;
-v^(-1)-2+3w,	+v^(-1)w^(-2)+2w^(-2)-3w^(-1)+1,	0,	-v^(-1)+2v^(-1)w-v^(-1)w^2,	+v^(-1)-1-v^(-1)w+w;
-v^(-1)w^(-1)+uv^(-1)w^(-1)+w^(-1)-uw^(-1)+v^(-1)-uv^(-1)-1+u,	+v^(-1)w^(-3)-uv^(-1)w^(-3)-w^(-3)+uw^(-3)-v^(-1)w^(-2)+uv^(-1)w^(-2)+w^(-2)-uw^(-2)+v^(-1)w^(-1)-uv^(-1)w^(-1)-w^(-1)+uw^(-1)-v^(-1)+uv^(-1),	+1-v^(-1)w^2,	+v^(-1)w-uv^(-1)w,	-1+u-v^(-1)w+uv^(-1)w;
+v^(-1)w^(-1)-uw^(-1)-v^(-1)+u,	-v^(-1)w^(-3)+uw^(-3)+v^(-1)w^(-2)-uw^(-2)-v^(-1)w^(-1)+uw^(-1),	+1-v^(-1)w,	+v^(-1)-uv^(-1)w,	-v^(-1)+u;
